dynamic-cat-api/index.js

```diff
@@ -1,40 +1,49 @@
-const nodeHtmlToImage = require('node-html-to-image');
-// const SomeRandomCat = require('some-random-cat').Random;
-const fetch = require('node-fetch')
-const fs = require('fs');
-const path = require('path');
+const { BlobServiceClient } = require("@azure/storage-blob");
+const Jimp = require('jimp');
 
 module.exports = async function (context, req) {
     context.log('JavaScript HTTP trigger function processed a request.');
 
     // get text from route parameter
     var text = context.bindingData.text;
 
-    // get random cat image url
-    // const catObj = await SomeRandomCat.getCat()
-    // const url = catObj.url;
-    const urlResp = await fetch("https://api.thecatapi.com/v1/images/search?format=json", {
-        method: 'GET',
-        headers: {
-            'x-api-key': process.env.CAT_KEY
-        }
-    });
-    const catObj = await urlResp.json();
-    const url = catObj[0].url;
-
-    // templating html
-    const templateHtml = fs.readFileSync(path.join(__dirname, 'templates.html'), 'utf8');
-    const image = await nodeHtmlToImage({
-        html: templateHtml,
-        content: {
+    const buffer = await getImageFromStorage();
+    const font = await Jimp.loadFont(Jimp.FONT_SANS_32_WHITE);
+ 
+    const image = await Jimp.read(buffer).then(img => {
+        return img.scaleToFit(600,600)
+    }).then(img=>{
+        return img
+        .print(font,0,0, {
             text: text,
-            catUrl: url
-        }
-      });
+            alignmentX: Jimp.HORIZONTAL_ALIGN_CENTER,
+            alignmentY: Jimp.VERTICAL_ALIGN_BOTTOM
+          },
+          img.bitmap.width,
+          img.bitmap.height)
+        .getBufferAsync(Jimp.MIME_JPEG);  
+    });
 
     context.res = {
-        // status: 200, /* Defaults to 200 */
-        header: { "content-type": "image/png" },
+        header: { "content-type": "image/jpg" },
         body: image
     };
+}
+
+async function  getImageFromStorage()
+{    
+    const blobServiceClient = BlobServiceClient.fromConnectionString(process.env["BlobStorageConnectionString"]);

```

Comment:
@ganning127  @emsesc  did you know if you use process.env["BlobStorageConnectionString"] rather than env.BlobStorageConnectionString then you can add your values for local dev to `local.settings.json`

Reply:
@blueboxes did not know that, that's for the tip!